SmartPointers/shared_pointer.hpp

```diff
@@ -1,80 +1,134 @@
-//
-//  shared_pointer.hpp
-//  Pointer
-//
-//  Created by Srushti Desai on 12/27/23.
-//
+#include <utility>
+#include <iostream>
+#include <stdio.h>
 
-#ifndef shared_pointer_hpp
-#define shared_pointer_hpp
+template <typename T> class SharedPointer {
 
-using namespace std;
-#include <iostream>
-template<class T>
-class SharedPointer
-{
-private:
-    T* m_ptr;
-    unsigned int* m_count;
-public:
-     SharedPointer():m_ptr(nullptr), m_count(new unsigned int(0))
-    {
-        
+  public:
+    // Constructor
+    SharedPointer() : m_ptr(nullptr), count(new int(1)) {
+        std::cout << "Called Shared Constructor \n";
     }
 
-    ~SharedPointer(){
-        //cleanup();
-        
+    // Paramatized Constructor
+    SharedPointer(T* ptr) : m_ptr(ptr), count(new int(1)) {
+        std::cout << "called  Shared Paramatized Constructor \n";
     }
 
-    SharedPointer(T * ptr) : m_ptr(ptr), m_count(new unsigned int(1))
-        {
+    // Copy Constructor
+    SharedPointer(const SharedPointer& p) : m_ptr(p.m_ptr), count(p.count) { 
+        std::cout << "called  Shared Copy Constructor \n";
+        (*count)++; 
         }
+    
+    // Destructor
+    ~SharedPointer() { 
+        std::cout << " Shared Destructor \n";
+        destroy(); }
 
-    SharedPointer(const SharedPointer& sharedPtr){ // Copy constructor
-        this->m_ptr = sharedPtr.m_ptr;
-        this->m_count = sharedPtr.m_count;
-        if(nullptr != sharedPtr.m_ptr)
+    // Destroy the object
```

Comment:
Remove this comment as well.

You add comments when you need to convey more information than function-name.